### PR TITLE
feat: add respective `json_is` UDFs for JSON type

### PR DESCRIPTION
**Pull request by CookiePieWw**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)

#3686 #4515 #4230

## What's changed and what's your intention?

As title. Add following `json_is` UDFs: `json_is_null`, `json_is_array`, `json_is_object`, `json_is_int`, `json_is_float`, `json_is_string`, `json_is_bool`. Also rename `to_json` to `parse_json` and add some sqlness tests.

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: tests/cases/standalone/common/function/json/json_get.result [hunk 1]
```diff
@@ -196,3 +196,66 @@ DROP TABLE jsons;
...
+
+SELECT * FROM jsons WHERE json_get_string(j, 'a.b.c') = 'foo';
+
++--------------------------------------------------------------------------------------+-------------------------+
+| j                                                                                    | ts                      |
++--------------------------------------------------------------------------------------+-------------------------+
+| 40000001100000015000001d614000000110000001500000106240000001100000011000000363666f6f | 1970-01-01T00:00:00.003 |
++--------------------------------------------------------------------------------------+-------------------------+
+
+SELECT * FROM jsons WHERE json_get_bool(j, 'a.b.c') = true;
```
Comment (WenyXu) [inline tests/cases/standalone/common/function/json/json_get.result:245]: let's use `json_to_string`

File: src/common/function/src/scalars/json/json_is.rs [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    array,
+    "Checks if the input JSONB is a JSON array"
+);
+json_is!(
+    JsonIsObject,
+    object,
+    "Checks if the input JSONB is a JSON object"
+);
+
+mod tests {
```
Comment (WenyXu) [inline src/common/function/src/scalars/json/json_is.rs:133]:
```suggestion
#[cfg(test)]
mod tests {
```

**Review by WenyXu — APPROVED** [review]
Great! 🚀 LGTM

**Comment by sunng87** [comment]
Do you think it will be a good idea to have `jq` as a built-in function? I think it can cover most JSON operations so we don't have to implement and document them one-by-one

**Comment by CookiePieWw** [comment]
> Do you think it will be a good idea to have `jq` as a built-in function? I think it can cover most JSON operations so we don't have to implement and document them one-by-one

Yes, `jq` is quite powerful, but it returns an json type. The problem is that when we use UDFs, the binary data of json type are processed through datafusion and then the type info are erased since datafusion has no json type, then the servers would return plain binary data to clients.

We may have 2 choice here:
- let users use `json_to_string` function whenever using json type.
- or let users specify the type they want here.

BTW, databend provide `jq` function as well as `is_<type>` function and many other functions. Maybe it's for flexibility?

**Comment by paomian** [comment]
In my opinion, parse_json should be called by default when not performing any get path operation on a json column, rather than having to call this function manually for every query. Refer to https://www.postgresql.org/docs/current/datatype-json.html

rest LGTM

**Comment by WenyXu** [comment]
> In my opinion, parse_json should be called by default when not performing any get path operation on a json column, rather than having to call this function manually for every query. Refer to https://www.postgresql.org/docs/current/datatype-json.html
> 
> But I think this can be optimized later without blocking this pr merge.
> 
> rest LGTM

Both ways work(i.e., `INSERT INTO table VALUES (parse_json('{ "a": 1}'));`, `INSERT INTO table VALUES ('{ "a": 1}');`)

**Review by paomian — APPROVED** [review]

**Pull request merged**